stable/kube-slack/templates/deployment.yaml

```diff
@@ -41,10 +43,6 @@ spec:
             - configMapRef:
                 name: {{ template "kube-slack.fullname" . }}
           resources:
-      {{ if .Values.image.pullSecret }}
```

Comment:
@pierluigilenoci This is undoing your changes from #21915, which is probably not intended. Maybe you just need to merge the upstream master branch from `helm/charts`.

Reply:
This is undoing my changes from #21915 on purpose because was useless and even a broken solution. This new PR is a better approach to solve the problem.

Reply:
Ok. Why exactly was it broken? In my opinion it would be better to fix the pullSecret usage, instead of removing it, since somebody else might still need it.

Reply:
If you use my solution you get something like that
```
imagePullSecrets:
        - name: yoursecret {}
```
with an extra `{}` and I was not able to fix it because in this moment I don't have the time to eat my brain out.

Reply:
So I preferred to remove it in favor of a solution that worked for what my needs were. That is to make this chart compatible with the pod security policy. Not the best solution ever but at least the chart is not broken.